packages/core/src/config/load.ts

```diff
@@ -120,25 +135,106 @@ export function getLintRawConfigWithMergedContentByPriority(
   };
 }
 
-async function resolveExtends(lintConfig: LintRawConfig): Promise<LintRawConfig> {
-  if (!lintConfig.extends || !lintConfig.extends.length) return lintConfig;
+export function resolveNestedPlugins(
+  configPath: string,
+  pluginConfigPath: string,
+  plugin: string | Plugin,
+): string | Plugin | undefined {
+  if (!plugin) {
+    return;
+  }
+
+  if (!isString(plugin)) {
+    return plugin;
+  }
+
+  if (isAbsoluteUrl(plugin as string)) {
+    throw new Error(
+      `Error configuration format not detected (${plugin}). Nested plugin should be path to local file`,
+    );
+  }
+  const pluginPath = path.resolve(path.dirname(pluginConfigPath), plugin as string);
+
+  return path.resolve(path.dirname(configPath), pluginPath);
+}
+
+async function resolveApis(
+  apis: Record<string, Api>,
+  configPath: string = '',
+  resolve?: RawResolveConfig,
+): Promise<Record<string, Api>> {
+  const apiKeys = Object.keys(apis);
+  if (isEmptyArray(apiKeys)) {
+    return apis;
+  }
+
+  const resolvedApis: Record<string, Api> = {};
+  for (const apiKey of apiKeys) {
+    if (!apis[apiKey]?.lint?.extends) {
+      resolvedApis[apiKey] = apis[apiKey];
+    }
+    const lint = await resolveExtends(apis[apiKey].lint as LintRawConfig, configPath, resolve);
+    resolvedApis[apiKey] = {...apis[apiKey], lint};
+  }
+
+  return resolvedApis;
```

Comment:
This looks much terser to me:
```suggestion
return apis.map((api) => ({
  ...api,
  lint: resolveExtends(apis.lint, configPath, resolve),
}));
```

Also, you don't have to check `if (!apis[apiKey]?.lint?.extends) {` since this  check (and even better) already performs at  the very beginning of `resolveExtends` function, see [here](https://github.com/Redocly/openapi-cli/pull/632/files#diff-91233afeb25fea2c651ad22452d823830a0c6e8378deba81a1109b57cdae3fb6R188-R190)